pkg/sandbox/runx/impl/registry/registry.go

```diff
@@ -2,28 +2,38 @@ package registry
 
 import (
 	"context"
+	"os"
+	"path/filepath"
 
 	"go.jetpack.io/pkg/sandbox/runx/impl/download"
 	"go.jetpack.io/pkg/sandbox/runx/impl/fileutil"
 	"go.jetpack.io/pkg/sandbox/runx/impl/github"
 	"go.jetpack.io/pkg/sandbox/runx/impl/types"
 )
 
+var xdgInstallationSubdir = "jetpack.io/pkgs"
+
 type Registry struct {
 	rootPath fileutil.Path
 	gh       *github.Client
 }
 
-func NewLocalRegistry(rootDir string) (*Registry, error) {
-	rootPath := fileutil.Path(rootDir)
-	err := rootPath.EnsureDir()
+func NewLocalRegistry(ctx context.Context, githubAPIToken string) (*Registry, error) {
+	cacheDir, err := os.UserCacheDir()
 	if err != nil {
+		cacheDir = "~/.cache"
```

Comment:
The `~` won't expand if the path is used outside a shell. You probably want:

```suggestion
		cacheDir = os.Expandenv("$HOME/.cache")
```

Reply:
` os.UserCacheDir()` usually only errors if $HOME is empty. Will return error instead .